appmap/_implementation/configuration.py

```diff
@@ -55,147 +56,137 @@ def packages(self):
         return self._config['packages']
 
 
-def name_in_set(name, which):
-    return any(filter(lambda n: name.startswith(n + '.') or name == n,
-                      which))
-
-
-def class_in_set(class_name, which):
-    return name_in_set(class_name + '.', which)
-
-
-def function_in_set(fn, which):
-    class_name, fn_name = split_function_name(fn)
-    if class_name is None:
-        # fn isn't in a class, ignore it.
-        return False
-
-    ret = (class_in_set(class_name, which)
-           or name_in_set('%s.%s' %(class_name, fn_name), which))
-
-    logger.debug(('function_in_set, class_name %s'
-                  ' fnname %s'
-                  ' which %s'
-                  ' ret %s'),
-                 class_name,
-                 fn_name,
-                 which,
-                 ret)
-    return ret
+def splitname(obj):
+    """
+    Given an object that has a __module__ and __qualname__,
+    return a list of name components from both.
+    """
+    return obj.__module__.split('.') + obj.__qualname__.split('.')
 
 
-class ConfigFilter(Filter):
+def startswith(needle, haystack):
     """
-    During initialization, the ConfigFilter class reads the `package`
-    entries from the config file.  Based on these entries, it creates
-    a set of paths that should be included, and another set of paths
-    that should be excluded.
-
-    When checking to see if a function should be included,
-    ConfigFilter looks for the function in both sets.  If the function
-    is found in the included set, it's wrapped for instrumentation.
-    If it's found in the excluded set, it's used without modification.
-    If it isn't found in either set, ConfigFilter passes the function
-    on to the next Filter.
-
-    To determine whether a function is in one of the sets, the
-    fully-qualified name of the function is considered, including the
-    packages(s), module, and class that contain it.  As an example,
-    consider a function named `pkg1.pkg2.mod1.Class1.func`.  Any of
-    `pkg1`, `pkg1.pkg2.`, `pkg1.pkg2.mod1`, `pkg1.pkg2.mod1.Class1`,
-    or `pkg1.pkg2.mod1.Class1.func` will match this function (causing
-    it to be included or excluded, as appropriate).  Note that only
-    full names will match.  (This is achieved in the code by appending
-    `.` to the name retrieved from the config file.)
+    Check if needle is a prefix of haystack.
```

Comment:
What are the types of needle and haystack? Maybe they could use more descriptive names?

Reply:
Any iterable will work; what would you suggest? I thought they're pretty descriptive, and in any case, the function is simple enough that it's quite easy to read to see.

Reply:
The function is called `startswith`. In what sense would a `haystack` start with a `needle`? Maybe just giving an example of usage would clear up my confusion.

Reply:
It's a metaphor (perhaps with a bit of a stretch) for finding the needle in a haystack. `prefix` and `collection`? Are we bikeshedding yet?